### ISSUE TITLE
External not primitive types like chrono::DateTime and rust_decimal::Decimal are treated as components causing: Could not resolve pointer in Swagger doc. 

### ISSUE THREAD

**Issue by c1wren** [juhaku/utoipa#73]
I have some structs that contain types from chrono and rust_decimal. Those types give errors in the Swagger UI.

For example: 

Could not resolve reference: Could not resolve pointer: /components/schemas/DateTime does not exist in document
Could not resolve reference: Could not resolve pointer: /components/schemas/Decimal does not exist in document

What is the proper way to handle these types?

**Comment by juhaku** [comment]
The issue arises because the parsing logic only recognizes primitive types to the level that is supported by the OpenAPI schema. And anything other than primitive type is treated as `component` in `components/schemas/..` but because these types are not primitive types nor they are components like objects that creates an issue for the parser. 

At the moment these kind of types are not supported. This is indeed a bug I'll try to add possibility to override the type format as a very next thing in my list. 

1. I will add support to override the type with `#[component(..)]` attribute over the field
2. I will try to add support for automatic resolving for those chrono and rust_decimal types as well. probably with possibility to enable them with specific feature flags.

For time being unfortunately you need to stick with `number` or `String` types. I know this is frustrating but I'll try to get fix done today latest tomorrow.

**Comment by juhaku** [comment]
There is a PR now for supportin chrono types. However it needs some tuning since the Rust version is not yet 1.60 in Github runners. :cry:  

Regarding the Decimal type I cannot add default support for it since there is no way for to know in which format or type it should 
be presented in the OpenAPI schema by just looking the type itself. For this reason the `Decimal` type needs to be used with custom `#[component(...)]` attribute flag to define type type and possible format. I wil add this with another PR. 

The format would be something like:
```rust
#[derive(Component)]
struct Record {
    id: i32,
    #[component(type = String)]
    decimal: Decimal,
}
```

**Comment by juhaku** [comment]
After next release this situation can be solved with following methods:

1. Enable `chrono_types` or `chrono_types_with_format` (docs and REAMDE.md will have sufficient documentation about them) feature flag. These will allow you directly use chrono types in structs and they will be counted as String types in OpenAPI docs according to https://swagger.io/specification/#data-types. 
2. If the interpreted data type is not correct there is a possibility to override the type for the field. See the utoipa docs -> component derive once released. 

Basically it allows you to define the types as follows:
```rust
#[derive(Component)]
struct Post {
    id: i32,
    value: String,
    // this will override the type for the `Decimal` type and treat it as a String in OpenAPI spec
    #[component(value_type = String)] 
    rating: Decimal,
}
```
Any type can be overridden but only with primitive type or String type according to OpenAPI specification about datatypes (the link above). Generic types are not allowed within `value_type`.

**Comment by juhaku** [comment]
Also I added support for the rust_decimal as well. So now after next release rust decimal `Decimal` type will also work. It will be interpreted as String by default and type can be overridden with above mentioned `value_type` attribute.

**Comment by c1wren** [comment]
Awesome, thank you!

**Comment by Cheban1996** [comment]
**Warning**
If you read this and want use `component` will be attention, component was rename to `schema` - https://github.com/juhaku/utoipa/pull/243
